src/lib/rule2B.ts

```diff
@@ -54,7 +54,7 @@ export function rule2B(
       return computeTextAlternative(labelElem, {
         directLabelReference: true,
         inherited: context.inherited,
-      });
+      }).name;
```

Comment:
Doesn't returning just the name here mean we lose track of the `nodesUsed` and `rulesApplied` in the recursive call?

Reply:
I think this works because it's the same `context` reference that's passed into these functions, so even if `nodesUsed` / `rulesApplied` aren't returned here they have still been updated.

This test passes:
```
it('returns correct nodesUsed and rulesApplied sets for aria-labelledby references', () => {
    render(html`
      <div id="foo" aria-labelledby="bar">Hi</div>
      <div id="bar">Hello world</div>
    `, container);
    const elem1 = document.getElementById('foo')!;
    const elem2 = document.getElementById('bar')!;
    expect(computeTextAlternative(elem1)).toEqual({
      name: 'Hello world',
      nodesUsed: new Set<Node>([elem1, elem2, elem2.childNodes[0]]),
      rulesApplied: new Set<Rule>(['2B', '2F', '2G']),
    });
  });
```
Which I think shows that recursive calls aren't emptying the sets.